src/app/components/source/UserSecurity.js

```diff
@@ -0,0 +1,522 @@
+                      className="login__password-input"
+                      onChange={this.handleFieldChange.bind(this)}
+                      error={!this.state.errors.password}
+                      helperText={this.state.errors.password ? null : this.renderMessage('passwordError')}
+                      placeholder={this.renderMessage('passwordInput')}
+                    />
+                    : null
+                  }
+                  {this.state.twoFactorAuthentication ?
+                    <CardText style={cardTextAuthStyle}>
+                      <RaisedButton
+                        style={{ marginLeft: 'auto', marginRight: units(2) }}
+                        onClick={this.handleSubmitTwoFactorAuthentication.bind(this, false)}
+                        className="user-two-factor__enable-button"
+                        disabled={this.state.submitDisabled}
+                        label={
+                          <FormattedMessage id="userSecurity.disableTwofactor" defaultMessage="Disable" />
+                        }
+                      />
+                    </CardText>
+                    : null
+                  }
+                </CardText>
+                {this.state.showFactorAuthForm ?
+                  <div>
+                    <CardText style={cardTextAuthStyle}>
+                      <h3 style={subTitleStyle}>
+                        {
+                          <FormattedMessage
+                            id="userSecurity.downloadHeader"
+                            defaultMessage="Step 2: Download"
+                          />
+                        }
+                      </h3>
+                      <span style={{ lineHeight: '25px' }}>
+                        {
+                          <FormattedMessage
+                            id="userSecurity.downloadDescription"
+                            defaultMessage="You'll need a two-factor app, like Google Authenticator, on your smartphone to proceed:"
+                          />
+                        }
+                      </span>
+                      <Link to={appsUrls.apple} target="_blank" style={{ padding: '5px' }} >
+                        <img src={appsUrls.appleImage} alt="" />
+                      </Link>
+                      <Link to={appsUrls.play} target="_blank" style={{ padding: '5px' }}>
+                        <img src={appsUrls.playImage} alt="" />
+                      </Link>
+                    </CardText>
+                    <CardText style={cardTextAuthStyle}>
+                      <h3 style={subTitleStyle}>
+                        {
+                          <FormattedMessage
+                            id="userSecurity.qrcodeHeader"
+                            defaultMessage="Step 3: Scan"
+                          />
+                        }
+                      </h3>
+                      <span style={{ lineHeight: '25px' }}>
+                        {
```

Comment:
are these brackets needed? seems there are a few of them

Reply:
no I removed all unneeded brackets